src/steps/users/create-or-update-contact.ts

```diff
@@ -0,0 +1,56 @@
+import { BaseStep, Field, StepInterface } from '../../core/base-step';
+import { FieldDefinition, RunStepResponse, Step, StepDefinition } from '../../proto/cog_pb';
+
+import { baseOperators } from '../../client/constants/operators';
+import * as util from '@run-crank/utilities';
+
+/**
+ * Note: the class name here becomes this step's stepId.
+ @see BaseStep.getId()
+ */
+export class CreateOrUpdateContact extends BaseStep implements StepInterface {
+
+  protected stepName: string = 'Create or update an Iterable Contact';
+
+  protected stepType: StepDefinition.Type = StepDefinition.Type.ACTION;
+
+  protected stepExpression: string = 'create or update an iterable contact';
+
+  protected expectedFields: Field[] = [{
+    field: 'contact',
+    type: FieldDefinition.Type.MAP,
+    description: 'Where keys represent contact profile field names as represented in the Iterable API (including email, which is not technically a profile field).',
```

Comment:
- [ ] Heh.  Safe to remove `which is not technically a profile field` from the description, here.